### ISSUE TITLE
If CubeMap is present in the parameters, a cast error will occur

### ISSUE THREAD

**Issue by kakkou** [Unity-Technologies/com.unity.palette-switch#131]
If it is `MaterialProperty.PropType.Texture`, it is not necessarily `UnityEngine.Texture2D`, but the code that assumes that is not good.

https://github.com/Unity-Technologies/com.unity.material-switch/blob/f08075594deee4c7f028062b0d9e55cabbe93474/Editor/MaterialSwitchEditorUtility.cs#L115

**Comment by sindharta** [comment]
Thank you for your report. I assume you are having errors when using materials that use non-Texture2D textures ? CubeMap, for example ?
It would be great if you can attach the errors that you are having too.

**Comment by sindharta** [comment]
@kakkou This should be fixed in version `0.8.2-preview`.
Please let me know how it goes for you.

**Comment by kakkou** [comment]
Sorry for the delay in responding.

The error I was getting was as follows.
```
InvalidCastException: Specified cast is not valid.
Unity.MaterialSwitch.MaterialSwitchEditorUtility.CreateMaterialProperties (UnityEngine.Material[] materials) (at Library/PackageCache/com.unity.material-switch@0.8.1-preview/Editor/MaterialSwitchEditorUtility.cs:111)
Unity.MaterialSwitch.MaterialSwitchClipEditor.OnEnable () (at Library/PackageCache/com.unity.material-switch@0.8.1-preview/Editor/MaterialSwitchClipEditor.cs:115)
UnityEditor.Timeline.ClipInspector:OnEnable() (at Library/PackageCache/com.unity.timeline@1.6.4/Editor/inspectors/ClipInspector/ClipInspector.cs:266)
```

Then I changed to `0.8.2-preview` and the above error no longer occurs and I confirmed that it is fixed.
Thank you.
